### PR TITLE
Update to Uno 3.0

### PR DESCRIPTION
**Pull request by ebariche**
﻿## Description of Change

Update Prism.Uno to Uno 3.0

### PR Checklist

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/PrismLibrary/Prism?pullRequest=2164) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/PrismLibrary/Prism?pullRequest=2164) before we can accept your contribution.<br/><br/>:x: ebariche [sign now](https://cla.dotnetfoundation.org/PrismLibrary/Prism?pullRequest=2164)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/PrismLibrary/Prism?pullRequest=2164) it.</sub>

File: e2e/Uno/HelloUnoWorld.Wasm/HelloUnoWorld.Wasm.csproj [hunk 1]
```diff
@@ -45,11 +43,11 @@
     <!-- Note that for WebAssembly version 1.1.1 of the console logger required -->
     <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="1.1.1" />
     <PackageReference Include="Microsoft.Extensions.Logging.Filter" Version="1.1.1" />
-    <PackageReference Include="Uno.Microsoft.Xaml.Behaviors.Uwp.Managed" Version="2.0.1-uno.28" />
-    <PackageReference Include="Uno.UI" Version="2.3.0-dev.2" />
-    <PackageReference Include="Uno.UI.RemoteControl" Version="2.3.0-dev.2" Condition="'$(Configuration)'=='Debug'" />
-    <PackageReference Include="Uno.Wasm.Bootstrap" Version="1.2.0-dev.24" />
-    <PackageReference Include="Uno.Wasm.Bootstrap.DevServer" Version="1.2.0-dev.24" />
+    <PackageReference Include="Uno.Microsoft.Xaml.Behaviors.Uwp.Managed" Version="2.0.2-uno.74" />
+    <PackageReference Include="Uno.UI.WebAssembly" Version="3.0.5" />
```
Comment (agneszitte-nventive) [inline e2e/Uno/HelloUnoWorld.Wasm/HelloUnoWorld.Wasm.csproj:47]: @ebariche no need for Uno.UI + Uno.UI.WebAssembly at the time ?

Comment (dansiegel) [reply e2e/Uno/HelloUnoWorld.Wasm/HelloUnoWorld.Wasm.csproj:47]: The Uno.UI build props/targets are accessible transitively, so you still get the proper build props/targets. I would assume that you do not need both, however there are some analyzers and additional resources that I'm not sure about. Once you both have determined whether as a best practice you only need a single `Uno.UI.WebAssembly` reference or if you also need the explicit `Uno.UI` reference and this conversation is marked as resolved, we'll merge the PR.

Comment (ebariche) [reply e2e/Uno/HelloUnoWorld.Wasm/HelloUnoWorld.Wasm.csproj:47]: Thanks @dansiegel !

Comment (ebariche) [reply e2e/Uno/HelloUnoWorld.Wasm/HelloUnoWorld.Wasm.csproj:47]: @agneszitte-nventive, it shouldn't be required, the E2E app works correctly !

Our new project template also doesn't have Uno.UI :

https://github.com/unoplatform/uno/blob/master/src/SolutionTemplate/UnoSolutionTemplate/Wasm/UnoQuickStart.Wasm.csproj

Let me get a second opinion from Carl !

Comment (carldebilly) [reply e2e/Uno/HelloUnoWorld.Wasm/HelloUnoWorld.Wasm.csproj:47]: The change `Uno.UI` -> `Uno.UI.WebAssembly` is the **RIGHT CHANGE** for a Wasm app project.

Comment (agneszitte-nventive) [reply e2e/Uno/HelloUnoWorld.Wasm/HelloUnoWorld.Wasm.csproj:47]: Missed that but the documentation is updated also @ebariche & @dansiegel ;)
https://github.com/unoplatform/uno/blob/665e3ff7f3ea386ae4f4d73f70aaf8898ab91a8e/doc/articles/migrating-from-previous-releases.md

Comment (carldebilly) [reply e2e/Uno/HelloUnoWorld.Wasm/HelloUnoWorld.Wasm.csproj:47]: Doc: https://platform.uno/docs/articles/migrating-from-previous-releases.html

Comment (dansiegel) [reply e2e/Uno/HelloUnoWorld.Wasm/HelloUnoWorld.Wasm.csproj:47]: thanks for the confirmation @carldebilly

**Pull request merged**